### PR TITLE
Sort providers by partners first, then alphabetically by name

### PR DESCRIPTION
**Pull request by tortila**
Closes https://github.com/thegreenwebfoundation/admin-portal/issues/266.

The root cause of the bug seems to be `NULL` value comparison in SQL. The field lookup `partner__in=[1, 2, 3]` [in Django translates to](https://docs.djangoproject.com/en/4.1/ref/models/querysets/#in) the SQL query: `SELECT ... WHERE partner IN (1, 2, 3);`. This is problematic because in SQL NULL values are not comparable:
```
> SELECT 1 WHERE NULL IN ("", NULL);
Empty set (0.00 sec)
```

The QuerySet lookup was changed to [`isnull`](https://docs.djangoproject.com/en/4.1/ref/models/querysets/#isnull), which is handled like this in SQL:
```
> SELECT 1 WHERE NULL IS NULL;
+---+
| 1 |
+---+
| 1 |
+---+
1 row in set (0.00 sec)
```
I also fixed the test, it failed to catch this bug before because the data that was used for the test didn't use NULL values, just empty strings.

This is fix should be enough to resolve the issue, but I suggest that we look into cleaning the data, so that we don't get bitten in the future by different presentations of non-partner providers in the database.

### CODE REVIEW

**Comment by mrchrisadams** [comment]
@tortila when I remember I try to use _rebase, then merge_ so far when merging in PRs, and use `git rebase -i` to clean up commit history if it's messy.

It's not a hard requirement, but if you prefer "squash then merge" would you let me know the reasoning before doing so?

We don't have a real convention set yet, so there's space to set a convention. There's an issue below for capturing some notes so far in terms of coding standards and conventions:

https://github.com/thegreenwebfoundation/admin-portal/issues/23

File: apps/greencheck/tests/test_legacy_v2_api.py [hunk 1]
```diff
@@ -62,49 +62,32 @@ class TestDirectoryListingView:
...
+        # given: 9 providers, out of which 3 are partners
+        created_providers = []
         for _ in range(3):
-            hosting_provider_factory.create(partner="", showonwebsite=True)
-
-        res = legacy_views.fetch_providers_for_country("US")
+            p1 = hosting_provider_factory.create(partner="", showonwebsite=True)
+            p2 = hosting_provider_factory.create(partner="Partner", showonwebsite=True)
+            p3 = hosting_provider_factory.create(partner=None, showonwebsite=True)
+            created_providers.extend([p1, p2, p3])
```
[the remark this answers is not in the document]
Comment (tortila) [reply apps/greencheck/tests/test_legacy_v2_api.py]: Thanks for your feedback! I added a note to explain the test setup.

One more thing I want to add: I added the variables p1, p2, p3 to refer to the objects that are being created by the factory. Previously those were retrieved from the database after the test setup was done. It's a small change but keeps the testing setup clear
